gdal/alg/viewshed.cpp

```diff
@@ -180,7 +180,22 @@ inline static double CalcHeight(double dfZ, double dfZ2, GDALViewshedMode eMode)
  *
  * @param pProgressArg The callback data for the pfnProgress function.
  *
- * @param papszExtraOptions Future extra options. Must be set to NULL currently.
+ * @param papszExtraOptions List of options
+ *
+ * Options:
+ *
+ *   MINIMUM_HEIGHT=i
```

Comment:
the naming of this parameter, and the 0, 1, 2 logic, is not super natural and user friendly. I would rather suggest using MODE=NORMAL/MIN_TARGET_HEIGHT_FROM_DEM/MIN_TARGET_HEIGHT_FROM_GROUND.
And in the code, use a enum to distinguish those cases for code clarity.
As this functionality has not yet make its way into an official version, the signature of the function could be altered to have the mode as a standard parameter instead of using papszExtraOptions, but I don't have a strong opinion on this.

apps/gdal_viewshed.cpp should also offer that option, and doc/source/programs/gdal_viewshed.rst and autotest/utilities/test_gdal_viewshed.py should be updated accordingly